### PR TITLE
Add pylint checker rule

### PR DESCRIPTION
**Pull request by nikolaik**
We found it useful to lint usage of this pytest marker and maybe others do too.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -136,6 +136,30 @@ If ``--no-only`` has already been passed (perhaps by way of ``addopts`` in
     $ py.test --no-only --only
 
 
+Pylint checker
+--------------
+
+If you use pylint you can help avoid commiting stray mark.onlys. Enable the pylint checker in your plugins and enabled the `unexpected-focused` rule.
```
Comment (theY4Kman) [inline README.rst:142]: nit: spelling, wording
```suggestion
If you use pylint, you can help avoid committing stray `only` marks with the bundled plugin. Just enable the pylint checker in your plugins and enable the `unexpected-focused` rule.
```

File: pytest_only/checker.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+        if len(mark_only_nodes) < 1:
+            return
+
+        for node in mark_only_nodes:
+            self.add_message("unexpected-focused", node=node)
+
+
+def register(linter: PyLinter) -> None:
+    linter.register_checker(MarkOnlyCheckerChecker(linter))
```
Comment (theY4Kman) [inline pytest_only/checker.py:48]: nit: newline at EOF
```suggestion
    linter.register_checker(MarkOnlyCheckerChecker(linter))

```

**Review by theY4Kman — CHANGES_REQUESTED** [review]
ooo, this is fantastic, and a wonderful idea! I use a Live Template in PyCharm to expand `only` to:
```py
#XXX######################################################################################
@getattr(__import__('pytest').mark, 'only')
#XXX######################################################################################
```
... because it's really, really hard to miss when doing a `git diff` :P I really like the idea of bundling a lint rule.

I've only got some minor nits. I might also want to move `pytest_only.checker` to `pytest_only.ext.pylint` or something, just for organization's sake. And I might add an optional/extras package set to the `pyproject.toml` just to make the dependencies more concrete.

This is baller, though. Thank you so much for submitting it! <3

**Comment by nikolaik** [comment]
Nice, will update with your suggestions. i'll try to add a test too

**Comment by nikolaik** [comment]
Neat trick with the live template in pycharm, never tried those 💡

**Comment by theY4Kman** [comment]
I pushed a bunch of stuff to your branch. It always feels a little weird pushing to someone else's repo :P But it sure is convenient, and it doesn't ditch the commit author field (like pulling it into my own branch and merging straight to master/opening a separate PR might).

Anywhoozlewoozle, I did the restructuring to `pytest_only.ext.pylint`, added a bunch of tests (using [pytest-pylint](https://github.com/carsongee/pytest-pylint)), and also realized `@pytest.mark.only` on a function isn't the only way to apply marks, so I added support for class-level decorators and `pytestmark = ...` at class/module levels. I also added support for async function defs, just in case anyone uses it with pytest-asyncio et al (like moi).

I'd really like to know if the changes I made still work in your env before I merge, though.

**Comment by nikolaik** [comment]
Most excellent! Will test your changes tomorrow!

**Comment by nikolaik** [comment]
Very nice, works just as expected, bravo!  Minus the comment on python version dependency resolution

**Comment by theY4Kman** [comment]
Agh, sorry, I got married this weekend, and things have been hectic. That's my excuse :P I'll revisit this later tonight, and get it all released.

File: pyproject.toml [hunk 3]
```diff
@@ -20,11 +20,17 @@ classifiers=[
 ]
 
 [tool.poetry.dependencies]
-python = '>=3.5'
-pytest = '>=3.6'
+python = '>=3.6.2,<4'
+pytest = [
+    {version = '<7.1', python = '^3.6'},
+    {version = '>=7.1', python = '^3.7'},
```
[the remark this answers is not in the document]
Comment (theY4Kman) [reply pyproject.toml:26]: Oops! Yes!

**Comment by theY4Kman** [comment]
Thanks for that version constraint fix. I wish I knew how to include that installation path in the tests; right now, it's using poetry to install all deps, then uses pip to override whatever version of pytest it installed =/

Also, thank you so much for taking the time to open the pull request, and for thinking about other peeps who'll find this very handy :) Aaaand sorry for the delay in merging :P

**Comment by nikolaik** [comment]
Wowza, congrats @theY4Kman and cheers to you both 🥂

**Pull request merged**